src/TraceParserGen.Tests/ParserGenerationTests.cs

```diff
@@ -0,0 +1,288 @@
+            Assert.Contains("TraceEventParser", generatedContent);
+
+            Output.WriteLine("Successfully generated parser from manifest");
+
+            // Act - Step 2: Create and build a test console application
+            string testProjectDir = Path.Combine(OutputDir, "TestApp");
+            Directory.CreateDirectory(testProjectDir);
+
+            CreateTestConsoleApp(testProjectDir, outputCsPath);
+
+            // Act - Step 3: Build the test application
+            var buildExitCode = BuildTestApp(testProjectDir);
+            Assert.Equal(0, buildExitCode);
+
+            // Act - Step 4: Run the test application
+            var runExitCode = RunTestApp(testProjectDir);
+
+            // Assert - Step 4: Verify test app ran successfully (no crashes, no asserts)
+            Assert.Equal(0, runExitCode);
+
+            Output.WriteLine("Test completed successfully");
+        }
+
+        private int RunTraceParserGen(string exePath, string manifestPath, string outputPath)
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = exePath,
+                Arguments = $"\"{manifestPath}\" \"{outputPath}\"",
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true
+            };
+
+            Output.WriteLine($"Running: {startInfo.FileName} {startInfo.Arguments}");
+
+            using (var process = Process.Start(startInfo))
+            {
+                var output = process.StandardOutput.ReadToEnd();
+                var error = process.StandardError.ReadToEnd();
+                
+                process.WaitForExit();
+
+                if (!string.IsNullOrWhiteSpace(output))
+                {
+                    Output.WriteLine("STDOUT:");
+                    Output.WriteLine(output);
+                }
+
+                if (!string.IsNullOrWhiteSpace(error))
+                {
+                    Output.WriteLine("STDERR:");
+                    Output.WriteLine(error);
+                }
+
+                return process.ExitCode;
+            }
+        }
+
+        private void CreateTestConsoleApp(string projectDir, string generatedParserPath)
+        {
+            // Get the path to TraceEvent assembly - it's in the test project's output directory
+            // since we have a ProjectReference
+            string traceEventAssembly = Path.Combine(Environment.CurrentDirectory, "Microsoft.Diagnostics.Tracing.TraceEvent.dll");
+            
+            // Create the .csproj file
+            string csprojContent = $@"<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net462</TargetFramework>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include=""Microsoft.Diagnostics.Tracing.TraceEvent"">
```

Comment:
Rather than using a `<Reference>` tag, this should be a `<ProjectReference>` tag pointing at TraceEvent.csproj in the repo.  You'll need to implement this by using a relative path (e.g. ..\..\TraceEvent\TraceEvent.csproj).  This might not be the right relative path, but you can tell by finding the source code at runtime.  You'll need to do this because TraceEvent.dll won't be the only dependency needed.